cx/twitter/server.js

```diff
@@ -0,0 +1,277 @@
+  deleteSubscription();
+  await deleteWebhooksByUrl(targetUrl);
+  await registerWebhook();
+  registerSubscription();
+}
+
+function registerWebhook() {
+  return new Promise((resolve, reject) => {
+    request.post(
+        'https://api.twitter.com/1.1/account_activity/all/' + environmentName +
+            '/webhooks.json',
+        {oauth: twitterOAuth, form: {url: targetUrl}},
+        function(err, resp, body) {
+          if (err) {
+            console.error('Failed to register webhook: ' + err);
+            reject();
+          } else {
+            resolve();
+          }
+        });
+  });
+}
+
+function registerSubscription() {
+  request.post(
+      'https://api.twitter.com/1.1/account_activity/all/' + environmentName +
+          '/subscriptions.json',
+      {
+        oauth: twitterOAuth,
+      },
+      function(err, resp, body) {
+        if (err) {
+          console.error('Failed to register subscription: ' + err);
+        }
+      });
+}
+
+function listWebhooks(url) {
+  return new Promise((resolve, reject) => {
+    request.get(
+        'https://api.twitter.com/1.1/account_activity/all/webhooks.json',
+        {oauth: twitterOAuth}, function(err, resp, body) {
+          if (err) {
+            console.error('Failed to check webhooks: ' + err);
+            reject(err);
+          }
+          const environments = JSON.parse(body).environments;
+          if (Array.isArray(environments)) {
+            const environment = environments.find((element) => {
+              return element.environment_name === environmentName;
+            });
+            if (environment) {
+              const webhooks =
+                  environment.webhooks.filter((value, index, arr) => {
+                    return value.url === url;
+                  });
+              resolve(webhooks);
+            }
+            resolve([]);
+          }
+        });
+  });
+}
+
+function deleteWebhookById(webhookId) {
+  return new Promise((resolve, reject) =>{
+    request.delete(
+        'https://api.twitter.com/1.1/account_activity/all/' + environmentName +
+            '/webhooks/' + webhookId + '.json',
+        {oauth: twitterOAuth}, function(err, resp, body) {
+          if (err) {
+            console.error('Failed to delete webhook: ' + err);
+            reject(err);
+          }
+          resolve();
+        });
+  });
+}
+
+async function deleteWebhooksByUrl(targetUrl) {
+  const webhooks = await listWebhooks(targetUrl);
+  for (webhook of webhooks) {
+    if (webhook.id) {
+      await deleteWebhookById(webhook.id);
+    }
+  }
+}
+
+function deleteSubscription() {
+  request.delete(
+      'https://api.twitter.com/1.1/account_activity/all/' + environmentName +
+          '/subscriptions.json',
+      {oauth: twitterOAuth}, function(err, resp, body) {
+        if (err) {
+          console.error('Failed to delete subscriptions: ' + err);
+        }
+      });
+}
+
+app.post('/', async function(req, res) {
```

Comment:
Looks like this handler doesn't return any response. It might be better to make sure that it will always return a response appropriately.

E.g.,
https://github.com/GoogleCloudPlatform/dialogflow-integrations/blob/master/cx/spark/server.js#L50

Reply:
The integration should send a response after every message now.